### PR TITLE
PR ready distraction free feature

### PR DESCRIPTION
**Pull request by tayloraviets**
Signed-off-by: Taylor <tayloraviets@gmail.com>

---
Issue #584 "Distraction Free Settings"
---

**Important note**
Please note that only PrestoN is able to merge Pull Requests into master.

**Pull Request Type**
Please select what type of pull request this is:
- [ ] Bugfix
- [X] Feature Implementation

**Related issue**
#584 #590 

**Description**
This adds a "Distraction Free Settings" section to settings with the following 8 options:
-Hide Video Views
-Hide Video Likes/Dislikes
-Hide Channel Subscribers
-Hide Comment Likes
-Hide Recommended Videos
-Hide Popular Videos
-Hide Trending Videos
-Hide Live Chat

**Testing (for code that is not small enough to be easily understandable)**
Tested on windows through the UI using the dev_runner script and VS code.
Tested using both the local and invidious player.
Tried to test thoroughly however I am not fully aware of freetube's capabilities as of yet

**Desktop (please complete the following information):**
 - OS: Windows 10 64bit
 - OS Version: 10.0.18362 Build 18362
 - FreeTube version: N/A

**Additional context**
584 called for the hiding comments however since they are pre-collapsed I chose not to include this.

issues noticed during development:
1. subcomment likes do not display when using local API, this bug is present on develop

2. if autoplay is enabled before turning off recommended videos then a video autoplays even when no recommendations are visible. was not able to find a way to stop it that wasn't hacky and/or confusing for the user.

3. streams don't appear to work with the invidious server, this also happens on develop

4. people on youtube who exclusively stream don't appear to have subscribers visible at all (example channel "stocks rocks") which displays as 0 subscribers on their channel page in freetube, technically a visual bug

5. when searching for a channel using the local API subCount isn't populated displaying as 0 subs, this is present in development

### CODE REVIEW

File: src/renderer/components/ft-list-channel/ft-list-channel.js [hunk 1]
```diff
@@ -40,7 +43,11 @@ export default Vue.extend({
       this.thumbnail = this.data.avatar
       this.channelName = this.data.name
       this.id = this.data.channel_id
-      this.subscriberCount = this.data.followers.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
+      if (this.hideChannelSubscriptions) {
+        this.subscriberCount = null
+      } else {
+        this.subscriberCount = this.data.followers.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
```
Comment (tayloraviets) [inline src/renderer/components/ft-list-channel/ft-list-channel.js:49]: this line is the cause of my listed issue number 5. followers doesn't appear to be an accessible field in this scope

**Comment by tayloraviets** [comment]
working on fixing linter issues

**Comment by GilgusMaximus** [comment]
PrestoN has to check your code. I am not familiar enough with adding these things to the code base.
However if you want: Could you create issues for the problems you found? If you do not want to, I can create them as well.

**Comment by tayloraviets** [comment]
@GilgusMaximus I will create issues for the problems I found (assuming they don't exist already) in a few hours when I get off work

File: static/locales/en-US.yaml [hunk 2]
```diff
@@ -194,6 +194,16 @@ Settings:
     Hide Videos on Watch: Hide Videos on Watch
     Fetch Feeds from RSS: Fetch Feeds from RSS
     Manage Subscriptions: Manage Subscriptions
+  Distraction Settings:
+    Distraction Settings: Distraction Free Settings
```
Comment (PrestonN) [inline static/locales/en-US.yaml:198]: Please have the object key match the English phrase.

**Review by PrestonN — CHANGES_REQUESTED** [review]
Hey there!  This looks really good actually.  This is more or less how I would have implemented this, so thank you very much!

Please take a look at the comments that I've made.  They're very small changes and should be quick to fix.

Also please fix the remaining linting issues that you have (Except for the one from App.js, that one isn't your fault).

**Comment by PrestonN** [comment]
I apologize.  I didn't address the issues you noticed.  I'll go ahead and do that.

1. We're aware of this.  We'll fix it in a future update.

2. This is a good point, and is something that I thought of as well.  We could probably just display a message that tells the user that the autoplay next video is disabled if they decide to also disable viewing the video recommendations.  You're welcome to include this functionality in this PR or I can work on it once this is merged.  Completely your choice.

3. and 4. Let's make a new issue about this so we can further investigate.

5. I'm aware of this.  Neither the local API or Invidious API returns a proper value, so I may remove this entirely.  You're welcome to remove it in this PR if you'd like.

**Comment by tayloraviets** [comment]
okay, changes made and linter happy. Sorry for all the commits, took a bit to realize I could just run the linter locally.

as for your comment @PrestonN I will now make issues for points 2, 3, and 4 of my additional context section. I'd normally try to fix the autoplay issue now but I expect to be busy the next few days.

File: src/renderer/store/modules/settings.js [hunk 3]
```diff
@@ -269,6 +308,30 @@ const actions = {
...
+              commit('setHideRecommendedVideos', result.value)
+              break
+            case 'hideTrendingVideos':
+              commit('setHideTrendingVideos', result.value)
+              break
+            case 'hidePopularVideos':
+              commit('setHidePopularVideos', result.value)
+              break
+            case 'hideLiveChat':
+              commit('setHideLiveChat, result.value')
```
Comment (PrestonN) [inline src/renderer/store/modules/settings.js:333]: This string needs to be fixed.

**Review by PrestonN — CHANGES_REQUESTED** [review]
Please make this last change and we should be good.  I'll handle the remaining issues in the next day or so.

**Comment by PrestonN** [comment]
Great!  Thank you so much for this pull request!  I know quite a few people are going to be pleased to see this. :)

Thanks again!

**Pull request merged**